### PR TITLE
Add configuration for readthedocs.org

### PR DESCRIPTION
**Pull request by sajith**
Issue is #5.

Normally the defaults of readthedocs.org's doc builders (Python 3.7 and Sphinx) should work, but in order to be able to build mflib API docs, we need a little more than that: we need Python 3.9, and some extra packages (furo, myst-parser).  This configuration should take care of that.

### CODE REVIEW

**Comment by sajith** [comment]
Status of this PR:

- I've created https://readthedocs.org/projects/fabrictestbed-mflib/. The docs, when they are successfully built, should appear at https://fabrictestbed-mflib.readthedocs.io/en/latest/.  
- The docs are currently being built (or: attempts are being made to build the docs) off my fork, off this specific PR branch.  (I'm using a fork because I can't push to this repo.  It is not strictly necessary that I should be able to push to this repo, I think.)
- Currently there's an error when building the docs at RTD, because networkx and networkx-query (they are fim dependencies) do not like each other:

```
Installing collected packages: snowballstemmer, pytz, mock, commonmark, charset-normalizer, zipp, urllib3, sphinxcontrib-serializinghtml, sphinxcontrib-qthelp, sphinxcontrib-jsmath, sphinxcontrib-jquery, sphinxcontrib-htmlhelp, sphinxcontrib-devhelp, sphinxcontrib-applehelp, Pygments, pillow, packaging, MarkupSafe, imagesize, idna, docutils, certifi, babel, alabaster, requests, Jinja2, importlib-metadata, sphinx, readthedocs-sphinx-ext, sphinx-rtd-theme, recommonmark
Successfully installed Jinja2-3.1.2 MarkupSafe-2.1.2 Pygments-2.14.0 alabaster-0.7.13 babel-2.11.0 certifi-2022.12.7 charset-normalizer-3.0.1 commonmark-0.9.1 docutils-0.18.1 idna-3.4 imagesize-1.4.1 importlib-metadata-6.0.0 mock-1.0.1 packaging-23.0 pillow-9.4.0 pytz-2022.7.1 readthedocs-sphinx-ext-2.2.0 recommonmark-0.5.0 requests-2.28.2 snowballstemmer-2.2.0 sphinx-6.1.3 sphinx-rtd-theme-1.2.0 sphinxcontrib-applehelp-1.0.4 sphinxcontrib-devhelp-1.0.2 sphinxcontrib-htmlhelp-2.0.1 sphinxcontrib-jquery-2.0.0 sphinxcontrib-jsmath-1.0.1 sphinxcontrib-qthelp-1.0.3 sphinxcontrib-serializinghtml-1.1.5 urllib3-1.26.14 zipp-3.13.0

[rtd-command-info] start-time: 2023-02-13T17:00:50.096804Z, end-time: 2023-02-13T17:01:58.571072Z, duration: 68, exit-code: 1
python -m pip install --upgrade --upgrade-strategy eager --no-cache-dir .[docs]
Processing /home/docs/checkouts/readthedocs.org/user_builds/fabrictestbed-mflib/checkouts/latest
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): still running...
  Preparing metadata (setup.py): finished with status 'error'
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [14 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/home/docs/checkouts/readthedocs.org/user_builds/fabrictestbed-mflib/checkouts/latest/setup.py", line 33, in <module>
          setup(
        File "/home/docs/checkouts/readthedocs.org/user_builds/fabrictestbed-mflib/envs/latest/lib/python3.9/site-packages/setuptools/__init__.py", line 152, in setup
          _install_setup_requires(attrs)
        File "/home/docs/checkouts/readthedocs.org/user_builds/fabrictestbed-mflib/envs/latest/lib/python3.9/site-packages/setuptools/__init__.py", line 147, in _install_setup_requires
          dist.fetch_build_eggs(dist.setup_requires)
        File "/home/docs/checkouts/readthedocs.org/user_builds/fabrictestbed-mflib/envs/latest/lib/python3.9/site-packages/setuptools/dist.py", line 806, in fetch_build_eggs
          resolved_dists = pkg_resources.working_set.resolve(
        File "/home/docs/checkouts/readthedocs.org/user_builds/fabrictestbed-mflib/envs/latest/lib/python3.9/site-packages/pkg_resources/__init__.py", line 777, in resolve
          raise VersionConflict(dist, req).with_context(dependent_req)
      pkg_resources.ContextualVersionConflict: (networkx 3.0 (/home/docs/checkouts/readthedocs.org/user_builds/fabrictestbed-mflib/checkouts/latest/.eggs/networkx-3.0-py3.9.egg), Requirement.parse('networkx<3.0,>=2.4'), {'networkx-query'})
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

Next steps: 

- Solve the above problem with building docs.  (It may need a fim release? I don't know for sure yet.)
- Add @cscarp as a project maintainer at https://readthedocs.org/projects/fabrictestbed-mflib/.
- When this PR is merged, switch repo at RTD to https://github.com/fabric-testbed/mflib, and default branch to `main`.
- A webhook may be need to be set up once this PR is merged.

**Comment by sajith** [comment]
Turns out that the problem with dependencies, stated above, can be worked around by installing _only_ the packages required to build docs.  Docs are now at https://fabrictestbed-mflib.readthedocs.io/en/latest/.

**Comment by sajith** [comment]
@cscarp Please review this and merge when ready!

After this PR is merged, there are a few more things needed to be done so that hosting docs on RTD can be considered "finished".  You should probably make an account on RTD (GitHub login will work) and join the project https://readthedocs.org/projects/fabrictestbed-mflib/, so that you can make the necessary changes.

**Review by cscarp — APPROVED** [review]
Adding requirements.txt for read the docs.

**Comment by cscarp** [comment]
Added readthedocs requirements.

**Pull request merged**
